lib/streamlit/runtime/connection_factory.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import importlib
+import os
+import re
+from datetime import timedelta
+from typing import Any, Dict, Type, TypeVar, overload
+
+from typing_extensions import Final, Literal
+
+from streamlit.connections import SQL, ExperimentalBaseConnection, Snowpark
+from streamlit.errors import StreamlitAPIException
+from streamlit.runtime.caching import cache_resource
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.runtime.secrets import secrets_singleton
+
+# NOTE: Adding support for a new first party connection requires:
+#   1. Adding the new connection name and class to this dict.
+#   2. Writing two new @overloads for connection_factory (one for the case where the
+#      only the connection name is specified and another when both name and type are).
+#   3. Updating test_get_first_party_connection_helper in connection_factory_test.py.
+FIRST_PARTY_CONNECTIONS = {
+    "snowpark": Snowpark,
+    "sql": SQL,
+}
+MODULE_EXTRACTION_REGEX = re.compile(r"No module named \'(.+)\'")
+MODULES_TO_PYPI_PACKAGES: Final[Dict[str, str]] = {
+    "MySQLdb": "mysqlclient",
+    "psycopg2": "psycopg2-binary",
+    "sqlalchemy": "sqlalchemy",
+    "snowflake": "snowflake-snowpark-python",
+    "snowflake.snowpark": "snowflake-snowpark-python",
+    "tenacity": "tenacity",
```

Comment:
nit: I thought this was added to the Streamlit core dependencies? Was it removed again?

Reply:
It just wasn't removed from this list as I did that in https://github.com/streamlit/streamlit/pull/6530, which hadn't been merged at the time